cranelift/codegen/src/isa/s390x/lower.isle

```diff
@@ -956,6 +956,15 @@
 (rule (lower (has_type (vr128_ty ty) (bnot x)))
       (vec_not ty x))
 
+;; With z15 (bnot (bxor ...)) can be a single instruction, similar to the
+;; (bxor _ (bnot _)) lowering.
+(rule 3 (lower (has_type (and (mie2_enabled) (fits_in_64 ty)) (bnot (bxor x y))))
+      (not_xor_reg ty x y))
+
+;; Combine a not/xor operation of vector types into one.
+(rule 4 (lower (has_type (vr128_ty ty) (bnot (bxor x y))))
```

Comment:
`vr128_ty` matches `$I128`, should this be lower priority than the new rule on line 1058?

Reply:
To confirm though, these are different architectures so I don't think there's overlap?

Reply:
Whoops! Sorry about that :)